spec/dmv_spec.rb

```diff
@@ -2,42 +2,158 @@
+    it 'Can return list of facilities offering a specified Service' do
+      @dmv.add_facilities([@facility_1, @facility_2, @facility_3])
 
-      @dmv.add_facility(@facility_1)
-      @dmv.add_facility(@facility_2)
-      @dmv.add_facility(@facility_3)
 
-      expect(@dmv.facilities_offering_service('Road Test')).to eq([@facility_2, @facility_3])
+      
+      @dmv.facilities[1].add_service('Vehicle Registration')
+      @dmv.facilities[2].add_service('Renew Drivers License')
+      @dmv.facilities[1].add_service('New Drivers License')
+      @dmv.facilities[1].add_service('Road Test')
+      @dmv.facilities[1].add_service('Written Test')
+      @dmv.facilities[2].add_service('Road Test')
+
+      @dmv.facilities[1].add_service('New Drivers License')
+      
+      
+
+      expect(@dmv.facilities_offering_service('Road Test')).to eq(@dmv.facilities[1..2])
+    end
+  end
+
+  describe '#format_name' do
+    it 'takes a given hash argument and returns a name string' do
+      
+      expect(@dmv.format_name(@facility_1)).to eq("Albany DMV Office")
+      expect(@dmv.format_name(@facility_2)).to eq('Ashland DMV Office')
+      expect(@dmv.format_name(@facility_3)).to eq('Bend DMV Office')
+    end
+
+    it 'takes a diffrent value from the hash argument' do
+
+      expect(@dmv.format_name(@or_locations[1])).to eq("Ashland DMV Office")
+      expect(@dmv.format_name(@ny_locations[1])).to eq("Rochester Downtown")
+      expect(@dmv.format_name(@mo_locations[1])).to eq("Bonne Terre")
+
+    end
+  end
+
+  describe '#format_address' do
+    it 'takes a hash and returns a string' do
+
+      expect(@dmv.format_address(@facility_1)).to eq("2242 Santiam Hwy SE Albany OR 97321")
+      expect(@dmv.format_address(@facility_2)).to eq('600 Tolman Creek Rd Ashland OR 97520')
+      expect(@dmv.format_address(@facility_3)).to eq('63030 O B Riley Rd Bend OR 97701')
+
+    end
+
+    it 'can take diffrent values from the hash and formate them' do
+
+      expect(@dmv.format_address(@or_locations[1])).to eq("600 Tolman Creek Rd Ashland OR 97520")
+      expect(@dmv.format_address(@ny_locations[1])).to eq("200 E Main Street Ste. 101 Rochester NY 14604")
+      expect(@dmv.format_address(@mo_locations[1])).to eq("30 N Allen St Bonne Terre MO 63628")
+    end
+  end
+
+  describe '#format_phone' do
+    it 'takes a phone string and formats' do
+
+
+      expect(@dmv.format_phone(@facility_1)).to eq('541-967-2014')
+      expect(@dmv.format_phone(@facility_2)).to eq('541-776-6092')
+      expect(@dmv.format_phone(@facility_3)).to eq('541-388-6322')
+
+    end
+
+    it 'can format from a diffrent source' do
+
+      expect(@dmv.format_phone(@or_locations[1])).to eq('541-776-6092')
+      expect(@dmv.format_phone(@ny_locations[1])).to eq("585-753-1604")
+      expect(@dmv.format_phone(@mo_locations[1])).to eq("573-358-3584")
+      
+    end
+  end
+  
+  describe '#format_hours' do
+    it 'takes hours from a facility hash' do
+      
     end
```

Comment:
Nice job testing all of the helper methods that you created. As I mentioned in the `dmv.rb` class I would create a `facility_factory_spec.rb` and move all of these tests in there. You would also create a `facility_factory.rb` and move all the methods you created for creating new facilities in there.